### PR TITLE
Fix part of #210: Add tests for TextInput's equals rule classifier 

### PR DESCRIPTION
**Pull request by miaboloix**
## Explanation

Fix part of #210
Added tests for TextInput's equals rule classifier
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [TextInputEqualsRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class TextInputEqualsRuleClassifierProviderTest {
+  private val STRING_VALUE_0 = createString(value = "TEST")
```
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt]: I suggest trying to capture the value in the constant name to provide more context at test sites, e.g.:

```kotlin
STRING_VALUE_TEST_LOWERCASE
STRING_VALUE_TEST_UPPERCASE
...
```

Ditto for each of these constants. Consider that we want at test sites to not have to leave the test body in order to understand, at a high level, what's being tested, and to verify that the test is correct.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @miaboloix! This looks good. I had just one comment on constant naming and would like to do one more pass on the PR once that's addressed.

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt (hunk 1 above)
Comment (miaboloix) [reply domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt]: Will do!

File: domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt [hunk 2]
```diff
@@ -22,13 +22,13 @@ import kotlin.test.fail
...
-  private val STRING_VALUE_2 = createString(value = "string")
-  private val STRING_VALUE_3 = createString(value = "test  a  lot  ")
-  private val STRING_VALUE_4 = createString(value = "test a lot")
-  private val STRING_VALUE_5 = createString(value = "testalot")
-  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 1)
+  private val STRING_VALUE_TEST_UPPERCASE = createString(value = "TEST")
+  private val STRING_VALUE_TEST_LOWERCASE = createString(value = "test")
+  private val STRING_VALUE = createString(value = "string")
+  private val STRING_VALUE_TEST_EXTRA_SPACES = createString(value = "test  a  lot  ")
+  private val STRING_VALUE_TEST_SINGL_SPACES = createString(value = "test a lot")
```
Comment (BenHenning) [inline domain/src/test/java/org/oppia/domain/classify/rules/textinput/TextInputEqualsRuleClassifierProviderTest.kt]: Nit(typo): ``STRING_VALUE_TEST_SINGLE_SPACES``

**Review by BenHenning — APPROVED** [review]
Thanks @miaboloix! I just had one comment. Feel free to mark it as resolved yourself once it's fixed, then the PR can be merged.

**Pull request merged**